File: lang/go/test/fun/vac/data_structures/LinkedStack/LinkedStackTest.go
package main

import (
    "fun/vac/data_structures/LinkedStack"
    "fun/vac/util/TestRunner"
)

func testLinkedStack() bool {
    size := 8192

    stack := LinkedStack.New()

    for i := 1; i <= size; i++ {
        stack.Push(i)
    }

    if stack.Size() != size {
        return false
    }

    for i := size; i > 0; i-- {
        if stack.Peek() != i {
            return false
        }
        stack.Pop()
    }

    return stack.IsEmpty()
}

func main() {
    TestRunner.ParseTest(testLinkedStack())
}

File: archive/nodejs/muse/data_structures/LinkedStackTest.js
import * as TestRunner from '../util/TestRunner.js';
import { LinkedStack } from './LinkedStack.js';

const testLinkedStack = () => {
  const size = 8192;

  const stack = new LinkedStack();

  for (let i = 1; i <= size; i++) {
    stack.push(i);
  }

  if (stack.size() !== size) {
    return false;
  }

  for (let i = size; i > 0; i--) {
    if (stack.peek() !== i) {
      return false;
    }
    stack.pop();
  }

  return stack.isEmpty();
};

const main = () => {
  TestRunner.pick(testLinkedStack);
};

main();
